Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Use the official Ubuntu 24.04 image as the base image
+FROM ubuntu:24.04
+
+# Set the working directory
+WORKDIR /usr/src/app
+
+# Install necessary dependencies
+RUN apt-get update && apt-get install -y \
+    curl \
+    protobuf-compiler \
+    xz-utils
+
+# Copy the source code and flake.nix into the container
+COPY . /usr/src/app
+
+# Create /nix directory and set permissions
+USER root
+RUN mkdir -m 0755 /nix && chown root /nix
+
+# Create nixbld group and users
+RUN groupadd -g 30000 nixbld && \
+    for i in $(seq 1 10); do \
+        useradd -u $((30000 + i)) -g nixbld -G nixbld -m -s /bin/false nixbld$i; \
+    done
+
+# Install Nix and set up the environment
+RUN curl -L https://nixos.org/nix/install | sh -s -- --daemon && \
```

Comment:
Curious as to why nix is needed

Reply:
Based on description from https://github.com/aki-mizu/cdk/tree/main/crates/cdk-mintd
In order to build from source, running nix develop is necessary.

Reply:
nixos has a image that we could use instead of installing it manually but we dont actually need it to build. Most of the stuff in the flake is for development

Reply:
I thought using nix is to have consistent development and deployment environment.
If not using nix, then we have to sync the rust environment to this file from time to time. 
e.g. when upgrading from rust 1.82 to 1.83.
But there might be better way to handle this if not using nix.

Reply:
Yeah but its really not needed for docker builds, rust version can be set in the docker file if need be. It only adds more build steps which can be avoided

Reply:
We want to avoid setting rust version directly in docker file since it’s already set by the flake.